src/IB/IBFEMethod.cpp

```diff
@@ -929,33 +929,67 @@ IBFEMethod::spreadForce(const int f_data_idx,
+    Pointer<PatchHierarchy<NDIM> > hierarchy = d_use_scratch_hierarchy ? d_scratch_hierarchy : d_hierarchy;
+    const int ln = hierarchy->getFinestLevelNumber();
+    const auto f_scratch_data_idx = d_active_eulerian_data_cache->getCachedPatchDataIndex(f_data_idx);
+    // zero data.
+    Pointer<hier::Variable<NDIM> > f_var;
+    VariableDatabase<NDIM>::getDatabase()->mapIndexToVariable(f_data_idx, f_var);
+    auto f_active_data_ops = HierarchyDataOpsManager<NDIM>::getManager()->getOperationsDouble(f_var, hierarchy);
+    f_active_data_ops->resetLevels(ln, ln);
+    f_active_data_ops->setToScalar(f_scratch_data_idx, 0.0, /*interior_only*/ false);
 
     // Spread interior force density values.
     for (unsigned int part = 0; part < d_num_parts; ++part)
     {
         PetscVector<double>* X_ghost_vec = d_X_IB_ghost_vecs[part];
         PetscVector<double>* F_ghost_vec = d_F_IB_ghost_vecs[part];
-        d_active_fe_data_managers[part]->spread(f_data_idx,
+        d_active_fe_data_managers[part]->spread(f_scratch_data_idx,
                                                 *F_ghost_vec,
                                                 *X_ghost_vec,
                                                 FORCE_SYSTEM_NAME,
-                                                f_phys_bdry_op,
+                                                /*f_phys_bdry_op*/ nullptr,
                                                 data_time,
                                                 /*close_F*/ false,
                                                 /*close_X*/ false);
     }
 
+    // Deal with force values spread outside the physical domain.
+    if (f_phys_bdry_op)
+    {
+        f_phys_bdry_op->setPatchDataIndex(f_scratch_data_idx);
+        Pointer<PatchLevel<NDIM> > level = hierarchy->getPatchLevel(ln);
+        for (PatchLevel<NDIM>::Iterator p(level); p; p++)
+        {
+            const Pointer<Patch<NDIM> > patch = level->getPatch(p());
+            Pointer<PatchData<NDIM> > f_data = patch->getPatchData(f_scratch_data_idx);
+            f_phys_bdry_op->accumulateFromPhysicalBoundaryData(*patch, data_time, f_data->getGhostCellWidth());
+        }
+    }
+
     if (d_use_scratch_hierarchy)
     {
+        // harder case: we have a scratch index on the scratch hierarchy but
+        // no corresponding index on the primary hierarchy.
+        //
+        // unlike the other data ops, this is always on the primary hierarchy
+        auto f_primary_data_ops = HierarchyDataOpsManager<NDIM>::getManager()->getOperationsDouble(f_var, d_hierarchy);
+        f_primary_data_ops->resetLevels(ln, ln);
+        const auto f_primary_scratch_data_idx = d_primary_eulerian_data_cache->getCachedPatchDataIndex(f_data_idx);
+        // we have to zero everything here since the scratch to primary
+        // communication does not touch ghost cells, which may have junk
+        f_primary_data_ops->setToScalar(f_primary_scratch_data_idx, 0.0, /*interior_only*/ false);
```

Comment:
@boyceg this is what I was missing an hour ago - oops.

Reply:
OK, so accumulation from the ghost region to the interior is done separately for each structure?

Reply:
No - see circa line 950. We do the accumulation after looping over all parts.

Reply:
Do different parts have different scratch hierarchies?

Reply:
No, but they do presently use different scratch patch data indices - this PR was written before the overloads #858 were added to avoid using a scratch force data index inside `FEDataManager::spread`.

Reply:
Can you remind me --- is there one scratch hierarchy, or are there separate scratch hierarchies for each part?

Reply:
(As you can probably guess, I am worried about losing the ghost values.)

Reply:
There is just one scratch hierarchy managed by IBFEMethod. The FEDataManagers in `d_scratch_fe_data_managers` all store `Pointer`s to that hierarchy instead of the 'normal' one.

Reply:
That's a really good point. Even though we have some tests based on `ex8` in the test suite which this passes, lets be paranoid and add some more spreading tests (as described below) before proceeding.